src/pprof/ddprof_pprof.cc

```diff
@@ -202,22 +202,18 @@ DDRes pprof_aggregate(const UnwindOutput *uw_output,
   ddprof_ffi_Label labels[PPROF_MAX_LABELS] = {};
   size_t labels_num = 0;
 
-  // The max PID on Linux is approximately 2^22 (~4e6), which also limits TID
-  // cppcheck-suppress variableScope
-  char pid_str[7] = {};
-  // cppcheck-suppress variableScope
-  char tid_str[7] = {};
-
   // Add any configured labels.  Note that TID alone has the same cardinality as
   // (TID;PID) tuples, so except for symbol table overhead it doesn't matter
   // much if TID implies PID for clarity.
-  if (watcher->send_pid || watcher->send_tid) {
+  if (!watcher->suppress_pid || !watcher->suppress_tid) {
+    char pid_str[sizeof("536870912")] = {}; // reserve space up to 2^29 base-10
```

Comment:
Not sure that reducing scope of `pid_str` / `tid_str` is correct: `labels` will store a pointer on a variable that has gone out-of-scope.

Reply:
Wow, I missed that, thanks ! Indeed, cppcheck is not detecting this correctly. To be fair the design pattern is weird.

Reply:
Not urgent, though trying to see how we can avoid this issue. Perhaps we should adapt this in some way. Having a general storage object ?

Reply:
Ah, good call--I followed the cppcheck nag without thinking!